### PR TITLE
first application gateway documenation template

### PR DESCRIPTION
**Pull request by vhorne**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by vhorne** [comment]
@bmoore-msft Not sure what the issue is here. Any ideas?

**Comment by bmoore-msft** [comment]
The name of the parameters file was incorrect - I renamed, so CI can run...

File: ag-docs-wafv2/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+              "CookieBasedAffinity": "Disabled"
+            }
+          }
+        ],
+        "httpListeners": [
+          {
+            "name": "appGatewayHttpListener",
+            "properties": {
+              "FrontendIPConfiguration": {
+                "Id": "[concat(variables('applicationGatewayID'), '/frontendIPConfigurations/appGatewayFrontendIP')]"
```
Comment (bmoore-msft) [inline ag-docs-wafv2/azuredeploy.json]: remove concat, use resourceId

File: ag-docs-wafv2/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+              },
+              "FrontendPort": {
+                "Id": "[concat(variables('applicationGatewayID'), '/frontendPorts/appGatewayFrontendPort')]"
```
Comment (bmoore-msft) [inline ag-docs-wafv2/azuredeploy.json]: remove concat, use resourceId

File: ag-docs-wafv2/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+            }
+          }
+        ],
+        "requestRoutingRules": [
+          {
+            "Name": "rule1",
+            "properties": {
+              "RuleType": "Basic",
+              "httpListener": {
+                "id": "[concat(variables('applicationGatewayID'), '/httpListeners/appGatewayHttpListener')]"
```
Comment (bmoore-msft) [inline ag-docs-wafv2/azuredeploy.json]: remove concat, use resourceId

File: ag-docs-wafv2/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+              },
+              "backendAddressPool": {
+                "id": "[concat(variables('applicationGatewayID'), '/backendAddressPools/appGatewayBackendPool')]"
```
Comment (bmoore-msft) [inline ag-docs-wafv2/azuredeploy.json]: remove concat, use resourceId

File: ag-docs-wafv2/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+              },
+              "backendHttpSettings": {
+                "id": "[concat(variables('applicationGatewayID'), '/backendHttpSettingsCollection/appGatewayBackendHttpSettings')]"
```
Comment (bmoore-msft) [inline ag-docs-wafv2/azuredeploy.json]: remove concat, use resourceId

File: ag-docs-wafv2/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+              }
+            }
+          }
+        ],
+        "webApplicationFirewallConfiguration": {
+          "enabled": "[parameters('wafEnabled')]",
+          "firewallMode": "[parameters('wafMode')]",
+          "ruleSetType": "[parameters('wafRuleSetType')]",
+          "ruleSetVersion": "[parameters('wafRuleSetVersion')]",
+          "disabledRuleGroups": []
```
Comment (bmoore-msft) [inline ag-docs-wafv2/azuredeploy.json]: remove empty properties

File: ag-docs-wafv2/azuredeploy.json (hunk 5 above)
Comment (vhorne) [reply ag-docs-wafv2/azuredeploy.json]: done

**Comment by vhorne** [comment]
@bmoore-msft 
OK, this should be ready to go now.

File: ag-docs-wafv2/azuredeploy.json (hunk 6 above)
Comment (bmoore-msft) [reply ag-docs-wafv2/azuredeploy.json]: this still needs to be done...

**Pull request merged**